### PR TITLE
Add a reproducible workflow strategy compatible with soft-fork

### PR DESCRIPTION
**Pull request by tgilon**
Closes https://github.com/open-energy-transition/cookiecutter-project/issues/2.

## Changes proposed in this Pull Request

To make the soft-fork strategy compatible with the concepts of the `cookiecutter`, it is suggested that the necessary files be included in the main repository. With this structure it should be easy to kick out projects as soft-forks.

## Checklist

- [x] I tested my contribution locally and it works as intended.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `envs/environment.yaml`.
- [ ] Changes in configuration options are added in `config/config.default.yaml`.
- [ ] Changes in configuration options are documented in `doc/configtables/*.csv`.
- [ ] Sources of newly added data are documented in `doc/data_sources.rst`.
- [ ] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

**Comment by tgilon** [comment]
@siddharth-krishna  @daniel-rdt @yerbol-akhmetov As discussed last week, here is the PR

File: README.md [hunk 1]
```diff
@@ -13,6 +13,53 @@ SPDX-License-Identifier: CC-BY-4.0
...
+
+# Installation and Usage
+
+## 1. Installation
+
+Clone the repository including its submodules:
+
+    git clone https://github.com/open-energy-transition/{{repository}}
+
+You need [mamba](https://mamba.readthedocs.io/en/latest/) to run the analysis. Using `mamba`, you can create an environment from within you can run it:
```
Comment (daniel-rdt) [inline README.md:45]: Should we reference mamba or conda as default package manager? Or both?

File: README.md [hunk 2]
```diff
@@ -13,6 +13,53 @@ SPDX-License-Identifier: CC-BY-4.0
...
+* `rules`: includes all the `snakemake`rules loaded in the `Snakefile`
+* `resources`: will store intermediate results of the workflow which can be picked up again by subsequent rules (does not exist initially)
+* `results`: will store the solved PyPSA network data, summary files and plots (does not exist initially)
+* `scripts`: includes all the Python scripts executed by the `snakemake` rules to build the model
+
+# Installation and Usage
+
+## 1. Installation
+
+Clone the repository including its submodules:
```
Comment (daniel-rdt) [inline README.md:41]: Are there submodules?

File: README.md (hunk 1 above)
Comment (tgilon) [reply README.md:45]: I don't have a strong recommendation here.

Comment (siddharth-krishna) [reply README.md:45]: Good question. This is probably a good place to recommend one of conda/mamba/micromamba if there's a significant difference between them? (I don't have much experience, I've mostly used micromamba and it worked well for me.) If there isn't a strong reason to recommend one, perhaps we could link to the installation page of all 3 and let the user pick?

File: README.md [hunk 3]
```diff
@@ -13,6 +13,53 @@ SPDX-License-Identifier: CC-BY-4.0
...
+
+Clone the repository:
+
+    git clone https://github.com/open-energy-transition/{{repository}}
+
+You need [mamba](https://mamba.readthedocs.io/en/latest/) to run the analysis. Using `mamba`, you can create an environment from within you can run it:
+
+    mamba env create -f environment.yaml
+
+Activate `{{project_short_name}}` environment:
```
Comment (siddharth-krishna) [inline README.md:49]:
```suggestion
Activate the newly created `{{project_short_name}}` environment:
```
Btw, is there some instructions/automation somewhere to change the environment name when someone sets up a new project repo?

File: README.md [hunk 4]
```diff
@@ -13,6 +13,53 @@ SPDX-License-Identifier: CC-BY-4.0
...
+## 2. Run the analysis
+
+    snakemake -call
+
+This will run all analysis steps to reproduce results and build the report.
+
+To generate a PDF of the dependency graph of all steps `resources/dag.pdf` run:
+
+    snakemake -c1 dag
+
```
Comment (siddharth-krishna) [inline README.md:62]:
```suggestion
----

----

```
Just to visually separate the fork's readme from the upstream's :)

**Review by siddharth-krishna — COMMENTED** [review]
Thanks, Thomas! This is a nice idea. I've left some minor comments. I'm also wondering if the first paragraph should say something about the soft fork strategy and link to the handbook?

File: README.md (hunk 1 above)
Comment (tgilon) [reply README.md]: I've added a reference to both `micromamba`  and `conda`.

File: README.md (hunk 3 above)
Comment (tgilon) [reply README.md:49]: Sadly, no.

**Comment by tgilon** [comment]
> Thanks, Thomas! This is a nice idea. I've left some minor comments. I'm also wondering if the first paragraph should say something about the soft fork strategy and link to the handbook?

@siddharth-krishna  I've added a new paragraph in the README.

**Comment by tgilon** [comment]
@pz-max I've updated the default licence to MIT. @martacki Do you have any other suggestions ?

**Pull request merged**